govcd/nsxt_route_advertisement_test.go

```diff
@@ -0,0 +1,63 @@
+//go:build network || nsxt || functional || openapi || ALL
+// +build network nsxt functional openapi ALL
+
+/*
+ * Copyright 2022 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+func (vcd *TestVCD) Test_NsxtEdgeRouteAdvertisement(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointNsxtRouteAdvertisement)
+
+	org, err := vcd.client.GetOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	nsxtVdc, err := org.GetVDCByName(vcd.config.VCD.Nsxt.Vdc, false)
+	check.Assert(err, IsNil)
+	edge, err := nsxtVdc.GetNsxtEdgeGatewayByName(vcd.config.VCD.Nsxt.EdgeGateway)
+	check.Assert(err, IsNil)
+
+	// Make sure we are using a dedicated Tier-0 gateway (otherwise route advertisement won't be available)
+	edge.EdgeGateway.EdgeGatewayUplinks[0].Dedicated = true
```

Comment:
let's check if this change is needed and return edge gtw to the same state as the test started to use. This allows us to avoid another test clash if in the case will use or depend on `Dedicated`

Reply:
+1 to this

Make sure to restore this setting post-test because we have other tests that create additional NSX-T Edge Gateways using the same external network (T0 router). When it becomes dedicated - no other Edge Gateways can be created using the same external network. Other tests would fail.

I think even a `defer` cleanup function would be good to define here in case the test breaks in between.

Reply:
Fixed. Please check https://github.com/mikeletux/go-vcloud-director/blob/31e5d2910eb6936914e847d408c6440585279b2d/govcd/nsxt_route_advertisement_test.go#L27-L33